src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubBranchSourceListener.java

```diff
@@ -0,0 +1,56 @@
+package org.jenkinsci.plugins.github_branch_source;
+
+import com.google.inject.Inject;
+import hudson.Extension;
+import hudson.model.Item;
+import hudson.model.listeners.ItemListener;
+import java.util.logging.Logger;
+import jenkins.branch.OrganizationFolder;
+import jenkins.model.Jenkins;
+import jenkins.scm.api.SCMNavigator;
+
+/**
+ * Listens for updates to GitHub branch sources in order to auto-register web hooks
+ */
+@Extension
+public class GitHubBranchSourceListener extends ItemListener {
+    @Inject
+    private transient Logger logger;
+
+    public GitHubBranchSourceListener() {
+        Jenkins.getActiveInstance().getInjector().injectMembers(this);
+    }
+
+    @Override
+    public void onCreated(Item item) {
+    }
+
+    @Override
+    public void onDeleted(Item item) {
+    }
+
+    @Override
+    public void onUpdated(Item item) {
+        if (item instanceof OrganizationFolder) {
```

Comment:
This is a :bug: You want to check for `SCMSourceOwner`, otherwise `AbstractGitHubSCMSource` implementations will not be able to notify statuses when working _alone_ (as `SCMSource` of a `MultiBranchProject` without an `OrganizationFolder` in the mix), for example _Workflow Multibranch_ projects.